packages/unmock-node/src/backend/index.ts

```diff
@@ -29,168 +32,159 @@ const mHttp = (
+        headers,
+        hostname: host,
+        method,
+        path,
+        story: story.story,
+        user_id: userId ? userId : MOSES,
+        ...(signature ? {signature} : {}),
       };
-      if (second && typeof second === "function") {
-        (second as ((res: IncomingMessage) => void))(res);
-      } else if (third && typeof third === "function") {
-        (third as ((res: IncomingMessage) => void))(res);
+      const hash = v0(hashable, ignore);
+      const hasHash = persistence.hasHash(hash);
+      const makesNetworkCall = mode === Mode.ALWAYS_CALL_UNMOCK ||
+        (mode === Mode.CALL_UNMOCK_FOR_NEW_MOCKS && !hasHash);
+      snapshot({
+        hash,
+        host,
+        method,
+        path,
+      });
+      if (!makesNetworkCall) {
+        const response = persistence.loadResponse(hash);
+        Object.entries(response.headers).forEach(([k, v]) => {
+          res.setHeader(k, `${v}`);
+        });
+        res.write(response.body);
+        res.end();
+        doEndReporting(true)(hash, body, response.headers, !response.body ? [] : [Buffer.from(response.body)]);
+        return {
+          body,
+          hash,
+          headers,
+          host,
+          intercepted: true,
+          method,
+          path,
+          port: 443,
+          req,
+          res,
+        };
+      } else {
+        return {
+          body,
+          hash,
+          headers: unmockHeaders,
+          host: unmockHost,
+          intercepted: false,
+          method,
+          path: pathForFake || "",
+          port: parseInt(unmockPort, 10),
+          req,
+          res,
+        };
       }
-    };
-    const output = cb(
-      fake,
-      selfcall ? (second as ((res: IncomingMessage) => void)) : resp,
-    );
-    if (token) {
-      output.setHeader("Authorization", `Bearer ${token}`);
-    }
-    output.setHeader(UNMOCK_UA_HEADER_NAME, JSON.stringify("node"));
-    const protoWrite = output.write;
-    output.write = (d: Buffer, q?: any, z?: any) => {
-      data = d;
-      return protoWrite.apply(output, [d, q, z]);
-    };
-    return output;
-  };
+     },
+     doEndReporting(false),
+  );
 };
 
 export default class NodeBackend implements IBackend {
   public reset() {
-    fr.http.request = httpreq;
-    http.request = httpreqmod;
-    https.request = httpsreq;
-    https.request = httpsreqmod;
+    mitm.disable();
   }
   public initialize(
+    userId: string,
     story: { story: string[] },
     token: string | undefined,
     options: IUnmockInternalOptions,
   ) {
-    fr.http.request = mHttp(story, token, options, httpreq);
-    http.request = mHttp(story, token, options, httpreqmod);
-    fr.https.request = mHttp(story, token, options, httpsreq);
-    https.request = mHttp(story, token, options, httpsreqmod);
+    mitm = Mitm();
+    mitm.on("connect", (socket: any, opts: any) => {
+      if (opts.shouldBypass) {
```

Comment:
Wouldn't this be `isWhiteListed(socket......)`?
It would make `mHttp` also shorter.

Reply:
Sorry, I don't understand this.  Could you explain?

Reply:
Yup! Why do we need to add the `shouldBypass` when creating a request (in `passthrough.ts`)?
The request is made to the Unmock API - which is whitelisted by default - so I thought it would make sense to use e.g.:
```typescript
mitm.on("connect", (socket: any, opts: any) => {
  if (hostIsWhitelisted(whitelist, opts.host)) {
    socket.bypass();
  }
}
```
(assuming we update `hostIsWhitelisted` to only use 2 arguments).

Then we can also remove the matching part in `mHttp`.